07_deploy_masters.sh

```diff
@@ -50,15 +50,16 @@ popd
 
 echo "Master nodes active"
 
-NUM_LEASES=$(sudo virsh net-dhcp-leases baremetal | grep master | wc -l)
-while [ "$NUM_LEASES" -ne 3 ]; do
-  sleep 10
-  NUM_LEASES=$(sudo virsh net-dhcp-leases baremetal | grep master | wc -l)
+# Wait for master nodes to have a DNS record
+for i in $(seq 0 2); do
```

Comment:
Iterating over this number is not reliable at all in baremetal since the masters can have any name. You should iterate over what the ETCD SRV record has:

    host -t SRV "_etcd-server-ssl._tcp.$CLUSTER_NAME" | awk '{print $NF}'

Will give you the names of the machines, which then you can use host or dig to resolve in the same way as you did in line 58.

Reply:
@celebdor thanks! I updated the change to iterate over the records returned by _etcd-server-ssl._tcp.${CLUSTER_DOMAIN}